### PR TITLE
Fix metrics tests

### PR DESCRIPTION
**Pull request by averevki**
Changes:
- Fix assert on deep-metrics tests
- Increase wait for metrics scrape
- Make a fixture that checks metrics availability on the cluster to run before other fixtures
- Add explicit id's on deep-metrics tests

### CODE REVIEW

File: testsuite/tests/kuadrant/authorino/metrics/test_deep_metrics.py [hunk 1]
```diff
@@ -56,5 +56,5 @@ def test_deep_metrics(metric_name, metric_type, deep_metrics):
         lambda x: x["metric"]["evaluator_name"] == metric_name and x["metric"]["evaluator_type"] == metric_type
     )
 
-    assert metrics
+    assert len(metrics.metrics) > 0
```
Comment (pehala) [inline testsuite/tests/kuadrant/authorino/metrics/test_deep_metrics.py:59]: in which scenario can it be more than one?

Comment (averevki) [reply testsuite/tests/kuadrant/authorino/metrics/test_deep_metrics.py:59]: If one of the AuthConfig sections was hit two times

**Comment by averevki** [comment]
^rebase

**Comment by martinhesko** [comment]
Tested this locally, ServiceMonitor gets deleted as expected, all the changes seem reasonable, LGTM on my side

**Review by martinhesko — APPROVED** [review]

File: testsuite/openshift/metrics/prometheus.py [hunk 2]
```diff
@@ -73,7 +74,7 @@ def wait_for_scrape(self, target_service: str, metrics_path: str = "/metrics"):
         """Wait before next metrics scrape on service is finished"""
         call_time = datetime.utcnow()
 
-        @backoff.on_predicate(backoff.constant, interval=10, jitter=None, max_tries=10)
+        @backoff.on_predicate(backoff.constant, interval=10, jitter=None, max_tries=20)
```
Comment (pehala) [inline testsuite/openshift/metrics/prometheus.py:77]: Do we really need this? and if so, do we know why it needs to be that big, what is causing it to be that horribly slow?

**Review by pehala — APPROVED** [review]

**Pull request merged**